lighthouse-core/lib/i18n/i18n.js

```diff
@@ -130,10 +130,66 @@ function lookupLocale(locale) {
  * @param {Record<string, string | number>} [values]
  */
 function _preprocessMessageValues(icuMessage, values = {}) {
-  const clonedValues = JSON.parse(JSON.stringify(values));
+  let clonedValues = JSON.parse(JSON.stringify(values));
   const parsed = MessageParser.parse(icuMessage);
+
+  const elements = _collectAllCustomElementsFromICU(parsed.elements);
+
+  clonedValues = _processParsedElements(Array.from(elements.values()), clonedValues);
+
+  return clonedValues;
+}
+
+/**
+ * Function to retrieve all 'argumentElement's from an ICU message. An argumentElement
+ * is an ICU element that has an argument attached like {varName, number, bytes}. This
+ * differs from 'messageElement's which are just arbitrary text in a message.
+ *
+ * Note: This function will recursively inspect plural elements for nested arguementElements.
+ *
+ * @param {Array<import('intl-messageformat-parser').Element>} elementsList
+ * @param {Map<string, import('intl-messageformat-parser').Element>} seenElelementsById
+ */
+function _collectAllCustomElementsFromICU(elementsList, seenElelementsById = new Map()) {
+  // add argumentElements
+  elementsList
+  .filter(el => el.type === 'argumentElement')
+  // @ts-ignore - el.id is always defined when el.format is defined
+  .forEach(el => seenElelementsById.set(el.id, el));
```

Comment:
since both this and the loop below only operate on type `argumentElement` (`pluralFormat` has to come from an `argumentElement`), can combine these two loops, something like
```js
for (const el of elements) {
  if (el.type !== 'argumentElement') continue;

  seenElelementsById.set(el.id, el);

  // Rescurse into Plurals
  if (!el.format || el.format.type !== 'pluralFormat') continue;

  // We need to find all the elements from the plural format sections, but
 ...
}
```

Reply:
GOOD CATCH!